### PR TITLE
Improve ProjectsController F=>A

### PR DESCRIPTION
**Pull request by calebthompson**
https://codeclimate.com/repos/54d4fa64695680762a0024c5/ProjectsController

See
https://codeclimate.com/repos/54d4fa64695680762a0024c5/compare/ct-lower-duplication-in-projects-controller#issues
for improvements, and individual commits for step-by-step embetterment.

* Remove before_actions to make up for gratuitous amounts of out-of-method
  config happening because of APIpie.
* Extract ProjectStaffController and ProjectApprovalsController as they
  were doing unrelated work and broke REST.
* Deduplicate params documentation, use with_options.

### CODE REVIEW

**Comment by chriskacerguis** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/54d4fa64695680762a0024c5/compare/ct-lower-duplication-in-projects-controller">analyzed this pull request</a>.

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -1,82 +1,72 @@
 class ProjectsController < ApplicationController
+  PROJECT_FIELDS = %w(alerts approvals approvers latest_alerts project_answers project_detail services staff)
```
Comment (stackus) [inline app/controllers/projects_controller.rb]: Might be better to name this `PROJECT_INCLUDES` to reflect the fact that it is a list of relationships.

File: app/controllers/project_approvals_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+class ProjectApprovalsController < ApplicationController
+  api :GET, '/projects/:project_id/approvals', 'Returns a list of all approvals for a project'
+  param :project_id, :number, required: true
+
+  def index
+    project = Project.find(params[:project_id])
```
Comment (stackus) [inline app/controllers/project_approvals_controller.rb]: Couldn't this by DRY'd up in a similar was as was done in `ProjectsController::project`?

File: app/controllers/project_staff_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+class ProjectStaffController < ApplicationController
+  api :GET, '/projects/:project_id/staff', 'Shows collection of staff for a :project_id'
+  param :project_id, :number, required: true
+  error code: 404, desc: MissingRecordDetection::Messages.not_found
+
+  def index
+    project = Project.find(params[:project_id])
```
Comment (stackus) [inline app/controllers/project_staff_controller.rb]: Same comment as the one from `ProjectApprovalsController::index`

**Comment by calebthompson** [comment]
@stackus updated.

**Comment by stackus** [comment]
:thumbsup: I really like the use of methods instead of `before_action`s too.

**Pull request merged**